edb/edgeql/compiler/policies.py

```diff
@@ -412,16 +420,29 @@ def try_type_rewrite(
         return
 
     # TODO: caching?
-    children_overlap = False
+    expand_children = False
+    exclude_cfg_views = False
     if children_have_policies:
         all_child_descs = [
             x
             for child in stype.children(schema)
             for x in child.descendants(schema)
         ]
+
+        # Children overlap
         child_descs = set(all_child_descs)
         if len(child_descs) != len(all_child_descs):
-            children_overlap = True
+            expand_children = True
+
+        # One of the children contains a cfg view.
+        if any(
+            irtyputils.is_excluded_cfg_view(
+                desc, ancestor=stype, schema=schema
+            )
+            for desc in all_child_descs
+        ):
+            expand_children = True
```

Comment:
Do we need to force the expansion here? If it falls back to relying on the `inheritance` expansion, that's fine?